node/task.ts

```diff
@@ -531,6 +532,25 @@ export function osType(): string {
     return os.type();
 }
 
+export enum Platform {
+    Windows,
+    MacOS,
+    Linux
+}
+
+/**
+ * Determine the operating system the build agent is running on.
+ * @returns Platform, or throws an `Error` if the platform is not recognized
+ */
+export function getPlatform(): Platform {
+    switch (process.platform) {
+        case 'win32': return Platform.Windows;
```

Comment:
According to the documentation (https://nodejs.org/api/process.html#process_process_platform) there are quite a few more strings that could be returned. Honestly, I don't know if we support some of them or not. I.e. AIX and SunOS might work but be unsupported. Perhaps the error message should indicate that the platform is not supported instead of not recognized.

Reply:
sure